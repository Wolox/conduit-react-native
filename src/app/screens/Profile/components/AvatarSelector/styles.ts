import { StyleSheet } from 'react-native';
import { green } from '@constants/colors';
import { scale } from '@utils/scalingUtils';
import { SPACING } from '@constants/dimentions';

const AVATAR_SIZE = scale(100);
const COMMON_POSITION = 0;
const COMMON_SPACING = 15;

export default StyleSheet.create({
  container: {
    height: '85%',
    width: '100%',
    alignItems: 'center',
    paddingBottom: SPACING.XXSMALL
  },
  avatarStyle: {
    width: AVATAR_SIZE,
    height: AVATAR_SIZE
  },
  itemStyle: {
    marginHorizontal: COMMON_SPACING,
    marginBottom: 10,
    padding: 7
  },
  selectedItem: {
    position: 'absolute',
    top: COMMON_POSITION,
    bottom: COMMON_POSITION,
    left: COMMON_POSITION,
    right: COMMON_POSITION,
    borderRadius: SPACING.XBIG,
    borderWidth: SPACING.XXSMALL,
    borderColor: green
  }
});
