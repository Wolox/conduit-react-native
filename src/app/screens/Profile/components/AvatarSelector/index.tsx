import React from 'react';
import { View, FlatList, ImageSourcePropType, Image, ListRenderItem, TouchableOpacity } from 'react-native';
import { useDispatch } from 'react-redux';
import { actionCreators as FeedbackActions } from '@redux/feedback/actions';
import { ListKeyExtractor } from '@interfaces/miscelanious';
import { avatarsIcons } from '@constants/iconsConstants';

import styles from './styles';

interface ItemAvatarSelector {
  name: string;
  icon: ImageSourcePropType;
}

interface AvatarSelectorProps {
  selected?: string;
  setSelected: (name: string) => void;
}

function AvatarSelector({ selected, setSelected }: AvatarSelectorProps) {
  const dispatch = useDispatch();
  const renderItem: ListRenderItem<ItemAvatarSelector> = ({ item }) => {
    const handleOnPress = () => {
      setSelected(item.name);
      dispatch(FeedbackActions.hideModal());
    };
    const isSelected = selected === item.name;
    return (
      <TouchableOpacity onPress={handleOnPress} style={styles.itemStyle}>
        <View style={isSelected && styles.selectedItem} />
        <Image source={item.icon} style={styles.avatarStyle} resizeMode="contain" />
      </TouchableOpacity>
    );
  };
  const keyExtractor: ListKeyExtractor<ItemAvatarSelector> = item => item.name;
  return (
    <View style={styles.container}>
      <FlatList data={avatarsIcons} renderItem={renderItem} numColumns={2} keyExtractor={keyExtractor} />
    </View>
  );
}

export default AvatarSelector;
