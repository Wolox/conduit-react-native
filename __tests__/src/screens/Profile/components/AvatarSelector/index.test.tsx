import React from 'react';
import renderer from 'react-test-renderer';
import AvatarSelector from '@screens/Profile/components/AvatarSelector';

describe('AvatarSelector', () => {
  test('AvatarSelector Snapshot', () => {
    const setSelected = (_name: string) => null;
    const item = { selected: 'Man1', setSelected };
    const tree = renderer.create(<AvatarSelector {...item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
