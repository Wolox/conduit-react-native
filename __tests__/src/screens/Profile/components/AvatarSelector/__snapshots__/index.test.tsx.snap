// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarSelector AvatarSelector Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingBottom": 4,
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Man1",
        },
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Man2",
        },
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Man3",
        },
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Man4",
        },
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Woman1",
        },
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Woman2",
        },
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Woman3",
        },
        Object {
          "icon": Object {
            "process": [Function],
          },
          "name": "Woman4",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#5CB85C",
                  "borderRadius": 32,
                  "borderWidth": 4,
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={false}
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={false}
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={false}
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={false}
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={false}
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={false}
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginBottom": 6,
                "marginHorizontal": 15,
                "opacity": 1,
                "paddingBottom": 12,
                "paddingHorizontal": 8,
                "paddingTop": 3,
              }
            }
          >
            <View
              style={false}
            />
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 214.28571428571428,
                  "width": 214.28571428571428,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
